gdal/ogr/ogrsf_frmts/gml/ogrgmldatasource.cpp

```diff
@@ -872,14 +872,24 @@ bool OGRGMLDataSource::Open( GDALOpenInfo *poOpenInfo )
                 {
                     GMLRegistryNamespace &oNamespace =
                         oRegistry.aoNamespaces[iNS];
-                    const char *pszNSToFind =
-                        CPLSPrintf("xmlns:%s", oNamespace.osPrefix.c_str());
-                    const char *pszURIToFind =
-                        CPLSPrintf("\"%s\"", oNamespace.osURI.c_str());
+                    // When namespace is omitted or fit with case sensitive match for
+                    // name space prefix, then go next to find feature match.
+                    //
```

Comment:
This change looks good.

I agree that the Open() method has grown unwieldy :-) Probably that part of the logic should be moved to smaller methods. But that would be things for another PR. Better not to mix several things at once.